src/TuxxAnimations.js

```diff
@@ -49,59 +50,323 @@ var createAnimationClass = function (transitions, customClassName) {
+                  enterActive['transform'] = enterActive['transform'].concat(' translateX(0px)');
+                }
+                //if origin contains 'bottom' then apply positive magnitude to translateY
+                if (setOrigin(origin,'bottom')) {
+                  //set transfrom properties
+                  enter['transform'] = enter['transform'].concat(' translateY(' + magnitude.toString() + 'px)');
+                  enterActive['transform'] = enterActive['transform'].concat(' translateY(0px)');
+                }
+                //if origin contains 'left' then apply negative magnitude to translateX
+                if (setOrigin(origin,'left')) {
+                  //set transfrom properties
+                  enter['transform'] = enter['transform'].concat(' translateX(-' + magnitude.toString() + 'px)');
+                  enterActive['transform'] = enterActive['transform'].concat(' translateX(0px)');
+                }
+              }//end enterProps.vector.origin
+
+              if (originType === 'leave') {
+                //if enterProps.vector.origin contains 'top' then apply negative magnitude to translateY
+                if (setOrigin(origin,'top')) {
+                  //set transfrom properties
+                  leave['transform'] = leave['transform'].concat(' translateY(0px)');
+                  leaveActive['transform'] = leaveActive['transform'].concat(' translateY(-' + magnitude.toString() + 'px)');
+                }
+                //if origin contains 'right' then apply positive magnitude to translateX
+                if (setOrigin(origin,'right')) {
+                  //set transfrom properties
+                  leave['transform'] = leave['transform'].concat(' translateX(0px)');
+                  leaveActive['transform'] = leaveActive['transform'].concat(' translateX(' + magnitude.toString() + 'px)');
+                }
+                //if origin contains 'bottom' then apply positive magnitude to translateY
+                if (setOrigin(origin,'bottom')) {
+                  //set transfrom properties
+                  leave['transform'] = leave['transform'].concat(' translateY(0px)');
+                  leaveActive['transform'] = leaveActive['transform'].concat(' translateY(' + magnitude.toString() + 'px)');
+                }
+                //if origin contains 'left' then apply negative magnitude to translateX
+                if (setOrigin(origin,'left')) {
+                  //set transfrom properties
+                  leave['transform'] = leave['transform'].concat(' translateX(0px)');
+                  leaveActive['transform'] = leaveActive['transform'].concat(' translateX(-' + magnitude.toString() + 'px)');
+                }
+              }//end leaveProps.vector.origin
+            } else {
+              throw 'magnitude property error: magnitude prop must have a number input written as {number}';
```

Comment:
Instead of throwing errors, use invariant. Look at other files in the tuxx/src for examples.